src/tests/builders/chains.ts

```diff
@@ -0,0 +1,118 @@
+import { faker } from '@faker-js/faker'
+import { RPC_AUTHENTICATION, GAS_PRICE_TYPE } from '@safe-global/safe-gateway-typescript-sdk'
+import type {
+  BlockExplorerUriTemplate,
+  ChainInfo,
+  GasPriceFixed,
+  GasPriceFixedEIP1559,
+  GasPriceOracle,
+  GasPriceUnknown,
+  NativeCurrency,
+  RpcUri,
+  Theme,
+} from '@safe-global/safe-gateway-typescript-sdk'
+
+import { Builder } from '@/tests/Builder'
+import { FEATURES } from '@/utils/chains'
+import { generateRandomArray } from './utils'
+import type { IBuilder } from '@/tests/Builder'
+import type useChains from '@/hooks/useChains'
+
+const rpcUriBuilder = (): IBuilder<RpcUri> => {
+  return Builder.new<RpcUri>()
+    .with('authentication', RPC_AUTHENTICATION.NO_AUTHENTICATION)
+    .with('value', faker.internet.url({ appendSlash: false }))
+}
+
+const blockExplorerUriTemplateBuilder = (): IBuilder<BlockExplorerUriTemplate> => {
+  return Builder.new<BlockExplorerUriTemplate>()
+    .with('address', faker.finance.ethereumAddress())
+    .with('txHash', faker.string.hexadecimal())
+    .with('api', faker.internet.url({ appendSlash: false }))
```

Comment:
This `with` is extremely boilerplaity. Why can't it just be an object?

Reply:
It can. I've changed it to accept `Partial<T>` instead.